opendse-io/src/main/java/net/sf/opendse/io/TGFFReader.java

```diff
@@ -0,0 +1,424 @@
+					importWiring(br, architecture);
+				}
+				
+				// import other @-annotated properties 
+				else if (!isComment(currentLine) && currentLine.contains(AT)) {
+					importProperty(currentLine);
+				}
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		} finally {
+			try {
+				if(br!=null){
+					br.close();
+				}
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+	
+	private void importCore(String name, BufferedReader br, Architecture<Resource, Link> architecture, Mappings<Task, Resource> mappings) throws IOException {
+		
+		// create resource (type)
+		Resource res = new Resource("r" + name.split(SEPARATOR)[1]);
+		
+		// first line contains attributes of resources
+		String [] resAttributes = (br.readLine()).replace(COMMENT, "").trim().split(SEPARATOR);		
+		
+		// second line contains attribute values
+		String [] resValues = br.readLine().trim().split(SEPARATOR);
+		
+		assert resAttributes.length == resValues.length : "tgff-file \"" + CORE + "\": number of values is not "
+				+ "equal to required number of resource attributes";
+		
+		for (int i = 0; i < resAttributes.length; i++) {
+			res.setAttribute(resAttributes[i], resValues[i]);
+		}			
+		architecture.addVertex(res);
+		
+		// create mappings
+		String line;
+		List<String> attributes = new LinkedList<String>();
+		
+    	while (!isClosing(line = br.readLine())) {
+		
+    		// extract attributes of resource type
+    		if (line.contains(TYPE)) {
+    			attributes = new LinkedList<String>(Arrays.asList(line.replace(COMMENT, "").trim().split(SEPARATOR)));
+    		}
+    		// extract values for each attribute
+    		else if (!isComment(line) && line.length() > 0) {
+
+    			String [] values = line.split(SEPARATOR);
+    			assert values.length == attributes.size() : "tgff-file \"" + CORE + "\": number of values is not "
+    					+ "equal to required number of attributes";
+    			
+    			String tgffType = values[0];
+    		
+    			// only add mappings to valid resource type
+    			boolean validMapping = false;
+    			
+    			if (attributes.contains(VALID)) {
+    				validMapping = values[attributes.indexOf(VALID)].equals("1") ? true : false;
+    			}
+    			
+    			
+    			// if tasks exist that can be mapped to current resource type
+    			if (validMapping && tgffTypeMap.containsKey(tgffType)) {
+		    			
+    				for (Task task : tgffTypeMap.get(tgffType)) {
+    					String mappingID = "m" + CONNECTOR + task.getId() + CONNECTOR + res.getId();
+    					Mapping<Task, Resource> mapping = new Mapping<Task, Resource>(mappingID, task, res);
+    					
+    					// annotate extracted attributes and values
+    					for (int i = 0; i < values.length; i++) {
+    						mapping.setAttribute(attributes.get(i), values[i]);
+    					}
+    					mappings.add(mapping);
+	    			}	
+    			}
+    		}		
+    	}		
+	}
+
+	private static Map<String, Double> importMessageSizes (BufferedReader br) throws NumberFormatException, IOException {

```

Comment:
Could you please add a comment explaining the intended usage of this method, especially why it needs to be static?